sdk/inc/azure/core/az_mqtt5_rpc_client.h

```diff
@@ -0,0 +1,334 @@
+
+/**
+ * @brief Triggers an AZ_EVENT_RPC_CLIENT_INVOKE_REQ event from the application
+ *
+ * @note This should be called from the application when wants to request a command to be invoked.
+ *
+ * @param[in] client The az_mqtt5_rpc_client_hfsm to use.
+ * @param[in] data The information for the execution request
+ *
+ * @return An #az_result value indicating the result of the operation.
+ * @retval #AZ_OK The event was triggered successfully.
+ * @retval #AZ_ERROR_HFSM_INVALID_STATE If called when the HFSM hasn't been asked to subscribe yet
+ * or is still subscribing.
+ * @retval #AZ_ERROR_NOT_SUPPORTED if the client is not connected.
+ * @retval Other on other failures creating/sending the request message.
+ */
+AZ_NODISCARD az_result az_mqtt5_rpc_client_invoke_req(
+    az_mqtt5_rpc_client_hfsm* client,
+    az_mqtt5_rpc_client_invoke_req_event_data* data);
+
+/**
+ * @brief Triggers an AZ_EVENT_RPC_CLIENT_SUB_REQ event from the application
+ *
+ * @note This should be called from the application to subscribe to the response topic. The RPC
+ * Client must be subscribed before commands can be invoked.
+ *
+ * @param[in] client The az_mqtt5_rpc_client_hfsm to use.
+ *
+ * @return An #az_result value indicating the result of the operation.
+ * @retval #AZ_OK The event was triggered successfully or the client is already subscribing.
+ * @retval #AZ_ERROR_HFSM_INVALID_STATE If called when the HFSM is already subscribed - the
+ * application doesn't need to wait for the AZ_EVENT_RPC_CLIENT_READY_IND event to start sending
+ * commands in this case.
+ * @retval #AZ_ERROR_NOT_SUPPORTED if the client is not connected.
+ * @retval Other on other failures creating/sending the subscribe message.
+ */
+AZ_NODISCARD az_result az_mqtt5_rpc_client_subscribe_req(az_mqtt5_rpc_client_hfsm* client);
+
+/**
+ * @brief Triggers an AZ_EVENT_RPC_CLIENT_UNSUB_REQ event from the application
+ *
+ * @note This should be called from the application to unsubscribe to the response topic. This will
+ * prevent the application from invoking commands unless it subscribes again. This may be used if
+ * the application doesn't want to recieve responses anymore.
+ *
+ * @param[in] client The az_mqtt5_rpc_client_hfsm to use.
+ *
+ * @return An #az_result value indicating the result of the operation.
+ * @retval #AZ_OK The event was triggered successfully.
+ * @retval #AZ_ERROR_NOT_SUPPORTED if the client is not connected.
+ * @retval Other on other failures creating/sending the unsubscribe message.
+ */
+AZ_NODISCARD az_result az_mqtt5_rpc_client_unsubscribe_req(az_mqtt5_rpc_client_hfsm* client);
+
+/**
+ * @brief Initializes an MQTT5 RPC Client HFSM.
+ *
+ * @param[out] client The az_mqtt5_rpc_client_hfsm to initialize.
+ * @param[in] connection The az_mqtt5_connection to use for the RPC Client.
+ * @param[in] property_bag The application allocated az_mqtt5_property_bag to use for the
+ * RPC Client.
+ * @param[in] client_id The client id to use for the response topic.
```

Comment:
Why do we need both client_id and executor_client_id?

Reply:
The response topic contains the client id of the invoker, and both the response and request topics contain the client id of the executor

Reply:
C# specifies the executor id per invocation, so it's on my todo list to see if that makes more sense for us as well

Reply:
Ah, I misunderstood what these meant. Maybe we could change the other one to "service_id" or "server_id" since we have RPCClient and RPCServer.
I do think it makes sense to have this on the request since the executor could be lightbulb1, lightbulb2, etc.

Reply:
Moved this to the request and called it server_client_id